### ISSUE TITLE
Python Types are missing :( 

### ISSUE THREAD

**Issue by Artur-Galstyan** [google-deepmind/mujoco#1292]
**Is your feature request related to a problem? Please describe.**
Pyright LSP cannot find the MjData, MjModel etc. types 😢 

**Describe the solution you'd like**
A potential solution could be adding python stubs, like a`mujoco.pyi` file or something?

![image](https://github.com/google-deepmind/mujoco/assets/63471891/d6e317b0-8ca0-4949-8a9a-def63b28347d)

FYI: I was checking out Brax when I noticed this

**Comment by kevinzakka** [comment]
Hi @Artur-Galstyan, I was able to use [pybind11-stubgen](https://github.com/sizmailov/pybind11-stubgen) and point Visual Code to the generated stubs to get type hints.

```
pip install pybind11-stubgen
pybind11-stubgen mujoco  # Installs stubs in $HOME/stubs
```

Now in visual code, I added the following to my `settings.json`:

```
"python.languageServer": "Pylance",
"python.analysis.stubPath": "~/stubs",
```

and the result:

<img width="1120" alt="Screenshot 2024-01-02 at 3 01 56 PM" src="https://github.com/google-deepmind/mujoco/assets/10518920/d2ffabaf-bb9d-4703-8abe-e639098768aa">

**Comment by mluogh-xdof** [comment]
I was able to get this to work using Pylance typechecking in CI as well. 

I had to patch the `_enum.pyi` file created because the function bindings created expect an int type which works in C because of the way enums are basically ints there, but the Python typechecker throws an error. So we have to patch them to be in IntEnums.

Here's the script I used for this and I put it in the `typings` folder in my project: https://gist.github.com/mluogh-xdof/805cd47daf7d3690495c74cb1141ddb7

I also ran it to generate bindings using Annotated because there were some typing errors with the standard numpy annotations due to dimension annotations (which might alternatively be fixed by this currently unmerged PR https://github.com/sizmailov/pybind11-stubgen/issues/264)

Now it doesn't just give typehints but gives reasonable typechecking and can be shared across teams/CI/etc.

tldr workflow:

In the root dir:

```
pybind11-stubgen mujoco -o typings/  --numpy-array-wrap-with-annotated
```

Then:

```
python typings/patch_mujoco_enums.py typings/mujoco/_enums.pyi
```

Then in `pyproject.toml`:

```
[tool.pyright]
stubPath = "typings"
```

EDIT: actually seems like a recent enough version mujoco is compiled with a version of `pybind11` that will use `typing.SupportsInt` and make this issue go away automatically: https://github.com/pybind/pybind11/pull/5540?

so this will work:
```
pybind11-stubgen mujoco -o ~/typings
```

**Comment by kevinzakka** [comment]
Thanks @mluogh-xdof (and hi @wuphilipp). FYI @saran-t

**Comment by GF-Huang** [comment]
Same for Pylance in VSCode.

<img width="1482" height="1123" alt="Image" src="https://github.com/user-attachments/assets/b6136762-af0a-461a-888a-5be08e192819" />

But works well in Pycharm.

<img width="1476" height="1170" alt="Image" src="https://github.com/user-attachments/assets/68665ccc-5b4b-4229-a945-4b386a74e571" />
